### PR TITLE
Add mtbundles schema for managed-tenants-bundles config.yaml

### PR DESCRIPTION
**Pull request by sugarraysam**
Add a schema for `config.yaml` file inside of managed-tenants-bundles.

Refactored a few constructs, and most importantly externalized the shared parts of `metadata`, `imageset` and `mtbundles` schemas as JSON pointers. (Could not get it to work with yaml...)

### CODE REVIEW

File: managedtenants/schemas/mtbundles.schema.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $ref: "shared/addon_parameters.yaml"
+      addOnRequirements:
+        $ref: "shared/addon_requirements.yaml"
+      subOperators:
+        $ref: "shared/sub_operators.yaml"
+      subscriptionConfig:
+        $ref: "shared/subscription_config.json"
+required:
+  - environments
+  - ocm
```
Comment (ashishmax31) [inline managedtenants/schemas/mtbundles.schema.yaml:29]: I don't think the ocm attribute is a required attribute

File: managedtenants/schemas/mtbundles.schema.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      enum:
+        - integration
+        - stage
+  ocm:
+    type: object
+    description: "Required OCM data for valid imagesets to be constructed by automated MRs to managed-tenants."
+    additionalProperties: false
+    properties:
+      addOnParameters:
+        $ref: "shared/addon_parameters.yaml"
```
Comment (ashishmax31) [inline managedtenants/schemas/mtbundles.schema.yaml:20]: Did you mean `.json` here?

**Comment by ashishmax31** [comment]
> Could not get it to work with yaml

Did some googling and it seems like we need an openAPI 3 parser to make it work for yaml.

File: managedtenants/schemas/mtbundles.schema.yaml (hunk 2 above)
Comment (sugarraysam) [reply managedtenants/schemas/mtbundles.schema.yaml:20]: nice catch

File: managedtenants/schemas/mtbundles.schema.yaml (hunk 1 above)
Comment (sugarraysam) [reply managedtenants/schemas/mtbundles.schema.yaml:29]: true, I suppose when we start using imagesets though it will become

**Comment by sugarraysam** [comment]
> Did some googling and it seems like we need an openAPI 3 parser to make it work for yaml.

Interesting. Do you have a link? Did we want to use it? I am personally fine /w JSON. WDYT?

**Comment by ashishmax31** [comment]
@sugarraysam I'm fine with json as well :)

**Review by ashishmax31 — APPROVED** [review]

**Pull request merged**